### PR TITLE
Update docker related doc

### PR DESCRIPTION
**Pull request by Milokita**
- [x] read the [CONTRIBUTING guidelines](README.md#contributing)
- [ ] raised a GitHub issue or discussed it on the projects chat beforehand
- [ ] added unit tests
- [ ] added integration tests
- [x] updated documentation if needed
- [x] updated CHANGELOG.md

### CODE REVIEW

File: docs/running-headscale-container.md [hunk 1]
```diff
@@ -85,6 +87,26 @@ Note: use `0.0.0.0:8080:8080` instead of `127.0.0.1:8080:8080` if you want to ex
...
+```yaml
+version: "3.7"
+
+services:
+  headscale:
+    image: headscale/headscale
+    restart: unless-stopped
+    container_name: headscale
+    ports:
+      - "8080:8080"
```
Comment (ohdearaugustin) [inline docs/running-headscale-container.md:101]: Set them default to 127.0.0.1. So people have explicitly make it public. I think this is the better default.

File: docs/running-headscale-container.md [hunk 2]
```diff
@@ -66,6 +66,8 @@ db_type: sqlite3
 db_path: /etc/headscale/db.sqlite
 ```
 
+Change the `unix_socket: /var/run/headscale/headscale.sock` to `unix_socket: ./headscale.sock` for gRPC socket if you encounter `Error starting server error="failed to set up gRPC socket` related error
```
Comment (ohdearaugustin) [inline docs/running-headscale-container.md:69]: This is currently discussed in #1417

**Review by ohdearaugustin — COMMENTED** [review]
We should first resolve #1417 and then see how we will do it.

File: docs/running-headscale-container.md (hunk 1 above)
Comment (Milokita) [reply docs/running-headscale-container.md:101]: agreed

File: docs/running-headscale-container.md (hunk 2 above)
Comment (Milokita) [reply docs/running-headscale-container.md:69]: Maybe we should introduce it as a temporarily fix for docker users?

Comment (ohdearaugustin) [reply docs/running-headscale-container.md]: This is now obsolete due to #1450.

**Review by ohdearaugustin — APPROVED** [review]

**Pull request merged**